ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/Contracts.kt

```diff
@@ -26,3 +27,29 @@ interface Contract {
  */
 @CordaSerializable
 interface Command
+
+/**
+ * Defines a verifiable command; that is, a command that implements its own contract verification logic.
+ */
+interface VerifiableCommand : Command {
+
+    /**
+     * Verifies the specified transaction associated with the current contract command.
+     *
+     * @param transaction The transaction to verify.
+     * @param signatories The signatories associated with the current contract command.
+     */
+    fun verify(transaction: UtxoLedgerTransaction, signatories: Iterable<PublicKey>)
```

Comment:
UtxoLedgerTransaction will contain the commands and signatories, so this is getting a bit circular. Perhaps we just pass in the self index in the list of commands